risc0/zkvm/src/host/client/env.rs

```diff
@@ -351,9 +352,9 @@ impl<'a> ExecutorEnvBuilder<'a> {
     }
 
     /// Set the path where segments will be stored.
-    pub fn segment_path<P: AsRef<Path>>(&mut self, path: P) -> &mut Self {
-        self.inner.segment_path = Some(path.as_ref().to_path_buf());
-        self
+    pub fn segment_path<P: AsRef<Path>>(&mut self, path: P) -> Result<&mut Self> {
+        self.inner.segment_path = Some(TempDir::new_in(path.as_ref())?.into_path());
```

Comment:
I think this breaks the user's intent. If the user wants to store segments in a particular directory, that feels to me like they're going to manage the lifetime of the segments.